org.adempiere.base/src/org/adempiere/base/ICreditManager.java

```diff
@@ -0,0 +1,30 @@
+/******************************************************************************
+ * Copyright (C) 2016 Logilite Technologies LLP								  *
+ * This program is free software; you can redistribute it and/or modify it    *
+ * under the terms version 2 of the GNU General Public License as published   *
+ * by the Free Software Foundation. This program is distributed in the hope   *
+ * that it will be useful, but WITHOUT ANY WARRANTY; without even the implied *
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.           *
+ * See the GNU General Public License for more details.                       *
+ * You should have received a copy of the GNU General Public License along    *
+ * with this program; if not, write to the Free Software Foundation, Inc.,    *
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA.                     *
+ *****************************************************************************/
+package org.adempiere.base;
+
+/**
+ * Interface for Credit Manager implementor
+ * 
+ * @author Logilite Technologies
+ * @since  June 25, 2023
+ */
+public interface ICreditManager
+{
+	/**
+	 * Check credit status as per document action
+	 * 
+	 * @param  docAction Document Action
+	 * @return           Error message
+	 */
+	public String creditCheck(String docAction);
```

Comment:
would be good for future extension of the API to return a class (for e.g CreditStatus) instead of just a String. Also, I think the method name of "checkCreditStatus" is probably more intuitive.

Reply:
I would like to understand more. What should his class hold other than CreditStatus?

Reply:
For now, just the error message. Perhaps, you can add a boolean flag as well. The primary idea here is you can expand this in future by adding more field to the CreditStatus class.